apps/aechannel/src/aesc_fsm.erl

```diff
@@ -3368,8 +3442,38 @@ account_type(Pubkey) ->
 
 check_accounts(Initiator, Responder) ->
     case {account_type(Initiator), account_type(Responder)} of
-        {{ok, basic}, {ok, basic}}  -> ok;
-        {{ok, generalized}, _}      -> {error, generalized_account};
-        {_, {ok, generalized}}      -> {error, generalized_account};
-        _                           -> {error, not_found}
+        {{ok, _}, {ok, _}}  -> ok;
+        _                   -> {error, not_found}
     end.
+
+combine_sigs_or_meta(HalfSigned1, HalfSigned2) ->
```

Comment:
I see that a correct ga auth fun [should enquire](https://github.com/aeternity/protocol/blob/34e8eeee3c9d3c7fafa83883bc7d95273315125d/generalized_accounts/ga_explained.md#authorization) [the inner tx hash](https://github.com/aeternity/aeternity/blob/996df3264e07c6b6f5b3f9b8f5c3d10ca2dd1ffe/apps/aega/src/aega_meta_tx.erl#L183), otherwise the auth can be used for forging other tx.

(Trying to understand how this combining works.)